monikeu/assignment2/src/main/java/TreeUtil.java

```
@@ -0,0 +1,78 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class solves basic assigment
+ * Assumptions:
+ * - Node is not an ancestor of itself (You can't be your grandma!)
+ */
+
+public class TreeUtil<T> {
+
+    public Node findANodeWithValue(BinTree tree, T value) {
+        if (value != null && tree != null) {
+            return searchForAKeyRecursively(tree.getRoot(), value);
+        }
+        return null;
+    }
+
+    public List<Node> findAncestorsOfNodeWithValue(BinTree tree, T value) {
+        List<Node> result = new ArrayList<>();
+        if (value != null) {
```

It's better to use "fail fast" approach to avoid complicated nested statements. Example:
```
  if (value == null) {
    return Collections.EmptyList();
  }
  Node node = findNode(value);
  if (node == null) {
    return ...
  }
```

This allows to better track unexpected behavior and read code easier.